File: Ex7/ex7.c
#include<stdio.h>
struct student {	// Structure Declaration
int rollno;
char name[20];
float sub[6];
float total,internal_mark[6];
};
void display(struct student stu[],int n);
int main()
{
struct student stu[20];	// Structure  variable Declaration
int i,j,k,n;
printf("\n Student Internal Mark Details");
printf("\n Enter the Number of Students : ");
scanf("%d",&n);
printf("\nRollNo\tName\n");	// Get student details
for(i=0;i<n;i++)
    {
scanf("%d%s",&stu[i].rollno,stu[i].name);
for(k=1;k<=5;k++)
        {
stu[i].total = 0.0;
for(j=1;j<=3;j++)
            {
printf("\n Enter the IA %d mark for sub %d:",j,k);
scanf("%f",&stu[i].sub[k]);
stu[i].total=stu[i].total+stu[i].sub[k];
            }
stu[i].internal_mark[k]=(stu[i].total/300)*20;
        }
    }
display(stu,n);
}
void display(struct student stu[],int n)
{
int i,j;
printf("\nThe Student Details are\n");
for(i=0;i<n;i++)
    {
printf("\nRollNo:%d\nName:%s\n", stu[i].rollno, stu[i].name);
for(j=1;j<=5;j++)
        {
printf("Subject%d - Internal Mark: %0.2f\n",j,stu[i].internal_mark[j]);
        }
    }
}

File: Ex7/ex7.py
from dataclasses import dataclass, field


@dataclass
class Student:
    roll_no: int
    name: str
    internal_marks: list[float] = field(default_factory=list)

    @property
    def total(self):
        return sum(self.internal_marks)


def main():
    students = []
    count = int(input("Enter the Number of Students : "))

    for _ in range(count):
        roll_no, name = input("Enter Roll No and Name: ").split(maxsplit=1)
        marks = []
        for subject in range(1, 6):
            subject_total = 0.0
            for ia in range(1, 4):
                mark = float(input(f"Enter the IA {ia} mark for sub {subject}: "))
                subject_total += mark
            internal_mark = (subject_total / 300) * 20
            marks.append(internal_mark)
        students.append(Student(int(roll_no), name, marks))

    print("\nThe Student Details are")
    for student in students:
        print(f"\nRollNo:{student.roll_no}\nName:{student.name}")
        for i, mark in enumerate(student.internal_marks, 1):
            print(f"Subject{i} - Internal Mark: {mark:.2f}")

if __name__ == "__main__":
    main()
